langmeta-sqlite/src/main/resources/semanticdb.ddl

```diff
@@ -0,0 +1,49 @@
+create table document(
+  id int,
+  filename text,
+  contents text,
+  language text,
+  primary key (id)
+);
+
+create table name(
+  id int,
+  document int,
+  start int,
+  end int,
+  symbol int,
+  is_definition boolean,
+  primary key (id),
+  foreign key (document) references document (id),
+  foreign key (symbol) references symbol (id)
+);
+
+create table message(
+  id int,
+  document int,
+  start int,
+  end int,
+  severity int,
+  text text,
+  primary key (id),
+  foreign key (document) references document(id)
+);
+
+create table symbol(
```

Comment:
I propose we split this into two tables

```sql
create table symbol(
  id int,
  symbol text,
)

create table denotation -- same as current symbol with FK on symbol for text
```

We can easily insert a reference to a symbol without worrying if a denotation exists or not. We can insert a symbol or ignore it on conflict http://www.sqlite.org/lang_conflict.html

Reply:
👍

Reply:
was btw thinking it's probably possible to effificently load the symbol table into memory using a radix tree, at least global symbols. That should avoid doing a roundtrip for inserts only to get the FK id on existing symbols.

Reply:
I don’t understand the proposal here.  Is denotation just a symbol that refers to a document?

Reply:
@olafurpg I think we could give a database-centric approach a try. If an insert is sub-millisecond even on insanely big codebases + whatever single or double-digit overhead that JDBC brings to the table, then what's the big deal in requiring an extra insert?

Reply:
What I think this pull request means is that we can try to write semantic tools in the same way how people write normal business apps - load up a database, send a few SQL queries, maybe do some updates and that's it. No compiler magic, no advanced data structures. It will be a huge breakthrough if it works, so how about we give it a try?